external/fv3fit/fv3fit/reservoir/config.py

```diff
@@ -35,6 +35,10 @@ class ReservoirHyperparameters:
         where all elements are sampled from random uniform distribution
         [-1, 1]. Changing this affects relative weighting of reservoir memory
         versus the most recent state.
+    ncols_state: If state is 2D (each column is a separate subdomain which uses
```

Comment:
Is it necessary for this and `state_size` to be passed as configuration/yaml arguments, or can they be determined automatically at train/run time from tensor shapes? This seems fairly difficult to configure and I don't see how it gets ignored for non-2D states.

Reply:
Removed, and instead `reset_state` now takes an arg `input_shape` which will provide this information when it is needed to initialize the reservoir state.